etl/sean.py

```diff
@@ -0,0 +1,68 @@
+'''
+Created on Dec 27, 2015
+@author: SeanEaster
+'''
+from math import floor
+
+
+import numpy as np
+import pandas as pd
+
+
+def standardize(array):
+    columnMeans = array.mean(axis=0)
+    return array - columnMeans
+
+# First, load the btc data and other assets
+btcDf = pd.DataFrame.from_csv('btc.csv')
+otherDf = pd.DataFrame.from_csv('data.csv')
+
+
+# running through different windows produces a lot of NaN values in rollingDeviations...this effs up np.divide later
+for _ in xrange(50):
+	STARTINDEX = np.random.randint(100)
+	LENGTH = np.random.randint(500)
+
+	btcValues = btcDf.ix[btcDf.index[STARTINDEX:(STARTINDEX + LENGTH)]].values.reshape((-1,))
+	meanBtcValues -= btcValues.mean()
+	btcStandardDev = meanBtcValues.std()
+
+	# Because numpy.correlate only supports single-dim arrays, we have to iterate
+	# Or I need to spend more time reading the docs. One of those two.
+
+	otherValues = otherDf.values
+	otherValues = standardize(otherValues)
+
+	rollingDeviations = pd.rolling_std(otherDf, LENGTH)#.values[(LENGTH - 1):, :]
```

Comment:
> ...start date and the end date and taking the standard deviation of this chunk

I don't think that's _quite_ it. What I think should be happening IIRC is that the start date is fixed, but end date is not. E.g. say your window is 30 days, and your start date three years ago. You want to search over the space of all possible 30-day windows beginning on or after that start date.

That's why you need the rolling standard deviations—you want to normalize each rolling sum-product by the local covariance over the window. Since the btc series (the 'template' if you want to think of it in [template matching](https://en.wikipedia.org/wiki/Cross-correlation#Normalized_cross-correlation) terms) is fixed, its standard deviation is too.

And I think the -1 was just to account for np arrays using zero-indexing, but time is long and memory fickle...